harper-core/src/linting/that_than.rs

```diff
@@ -0,0 +1,240 @@
+use crate::{
+    Token,
+    expr::{Expr, SequenceExpr},
+    linting::{ExprLinter, Lint, LintKind, Suggestion},
+};
+
+pub struct ThatThan {
+    expr: Box<dyn Expr>,
+}
+
+impl Default for ThatThan {
+    fn default() -> Self {
+        let adjective_er_that_nextword = SequenceExpr::default()
+            .then(|tok: &Token, src: &[char]| {
+                if !tok.kind.is_comparative_adjective() {
+                    return false;
+                }
+                let adj_lower = tok.span.get_content_string(src).to_lowercase();
+                let is_better_or_later = adj_lower == "better" || adj_lower == "later";
+                !is_better_or_later
+            })
+            .t_ws()
+            .t_aco("that")
+            .t_ws()
+            // .then_any_word();
+            .then(|tok: &Token, src: &[char]| {
+                if !tok.kind.is_word() {
+                    return false;
+                }
+                let adj_lower = tok.span.get_content_string(src).to_lowercase();
+                let is_way = adj_lower == "way";
+                !is_way
+            });
```

Comment:
Don't you have a need for a method like this that would be

.any_words_except("way", "foo", "bar)

Here I'm using the variadic common pattern I would use in Go,but I'm unsure it exists in Rust.

Reply:
> Don't you have a need for a method like this that would be
> 
> .any_words_except("way", "foo", "bar)
> 
> Here I'm using the variadic common pattern I would use in Go,but I'm unsure it exists in Rust.

I actually made one and in fact one is kinda made by a macro which makes lots of methods, but it was subtly different and I kept getting unexpected results, so I backed that all out and did it the simple straightforward way and then it worked first time.

Most of the `.then` methods are derived from `.is` methods that can come up from either `TokenKind` or `WordMetadata` or both, but you can also add custom ones at the various levels.